mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -420,41 +449,77 @@ def _get_datetimes(ds, timeVariableName, yearOffset):  # {{{
     02/16/2017
     """
 
-    if isinstance(timeVariableName, (tuple, list)):
+    if isinstance(inTimeVariableName, (tuple, list)):
         # we want to average the two
-        assert(len(timeVariableName) == 2)
-        starts = _get_datetimes(ds, timeVariableName[0], yearOffset)
-        ends = _get_datetimes(ds, timeVariableName[1], yearOffset)
-        datetimes = [starts[i] + (ends[i] - starts[i])/2
-                     for i in range(len(starts))]
-        return datetimes
-
-    timeVariable = ds[timeVariableName]
-
-    if timeVariable.dtype == '|S64':
-        # this is a variable like date strings like 'xtime'
-        time = [''.join(atime).strip() for atime in timeVariable.values]
-        datetimes = [datetime.datetime(yearOffset + int(x[:4]),
-                                       int(x[5:7]), int(x[8:10]),
-                                       int(x[11:13]), int(x[14:16]),
-                                       int(x[17:19]))
-                     for x in time]
-    elif timeVariable.dtype == 'float64':
-        # this array contains floating-point days like
-        # 'daysSinceStartOfSim'
-        start = datetime.datetime(year=yearOffset+1, month=1, day=1)
-        datetimes = [start + datetime.timedelta(x)
-                     for x in timeVariable.values]
-    elif timeVariable.dtype == 'timedelta64[ns]':
-        # this array contains a variable like 'daysSinceStartOfSim' as a
-        # timedelta64
-        start = datetime.datetime(year=yearOffset+1, month=1, day=1)
-        datetimes = [start + x for x in
-                     pd.to_timedelta(timeVariable.values, unit='ns')]
+        assert(len(inTimeVariableName) == 2)
+
+        dsStart = _parse_dataset_time(
+            ds=ds,
+            inTimeVariableName=inTimeVariableName[0],
+            calendar=calendar,
+            simulationStartTime=simulationStartTime,
+            outTimeVariableName=outTimeVariableName,
+            referenceDate=referenceDate)
+        dsEnd = _parse_dataset_time(
+            ds=ds,
+            inTimeVariableName=inTimeVariableName[1],
+            calendar=calendar,
+            simulationStartTime=simulationStartTime,
+            outTimeVariableName=outTimeVariableName,
+            referenceDate=referenceDate)
+        starts = dsStart[outTimeVariableName].values
+        ends = dsEnd[outTimeVariableName].values
+
+        # replace the time in starts with the mean of starts and ends
+        dsStart[outTimeVariableName] = [starts[i] +
+                                        (ends[i] - starts[i])/2
+                                        for i in range(len(starts))]
+
+        return dsStart
+
```

Comment:
I just meant we might want to have a comment to clarify because the logic is a little confusing, but this is really a non-issue and minor point.  Let's skip a change here for now.